server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java

```diff
@@ -189,8 +189,7 @@ public void clusterChanged(ClusterChangedEvent event) {
                         final RepositoriesMetadata repoMeta =
                             event.state().metadata().custom(RepositoriesMetadata.TYPE);
                         final RepositoryMetadata metadata = repoMeta.repository("test-repo");
-                        if (metadata.generation() == metadata.pendingGeneration()
-                            && metadata.generation() > snapshotEntry.repositoryStateId()) {
+                        if (metadata.pendingGeneration() > snapshotEntry.repositoryStateId()) {
```

Comment:
The concrete situation that this test was covering is technically gone with this change, but I think it's reasonable to keep testing the very similar situation of master fail-over during the repository side of the finalization here now.